lib/money/money.rb

```diff
@@ -603,6 +644,11 @@ def decimal_mark
 
   private
 
+  def dup_with_new_fractional(new_fractional, infinite_precision = nil)
+    infinite_precision = self.infinite_precision? if infinite_precision.nil?
+    self.class.new(new_fractional, currency, { bank: bank, infinite_precision: infinite_precision })
```

Comment:
nit: no need for curly braces when the last argument is a Hash

Reply:
Can totally remove them... though from your other comment it sounded like you wanted to move to keyword arguments?